daffodil-cli/src/it/scala/org/apache/daffodil/schematron/package.scala

```diff
@@ -94,7 +94,7 @@ package object schematron {
   private val mustache2 = """\{\{(.+?)}}""".r.unanchored
 
   private val echoEC = s"echo ${if(Util.isWindows) "%errorlevel%" else "$?"}"
-  private val matchEC = regexLine("""\d+""")
+  private val matchEC = regexLine(s"""(?<=\\d+)${if(Util.isWindows) eol else "$"}""")
```

Comment:
Look behind to find the so it is not consumed and is available in `getBefore`

Had to match the end of line differently, not entirely sure why `$eol` (`\n`) doesnt match on linux.

Reply:
Put this discussion into a comment in front of the offending line, maybe somebody will figure it out in the future. Probably CRLF vs. LF related. I think of '\n' as being LF, not CRLF. I think modern regex has \R which is \n or \r\n.